DOM.js

```diff
@@ -13,469 +13,494 @@
-         * @argument {DOMElement|DOMElement[]} args Element(s) to add as children
-         * @returns {DOMElement} self
-         */
-        DOMElement.prototype.addElem = function addComponent () {
-            var components = Array.prototype.slice.call(arguments);
-            Array.prototype.push.apply(this.myComponents, components);
-            return this;
-        };
-
-        /**
-         * Adds a text node
-         * @param icomp element
-         * @returns {DOMElement} self
-         */
-        DOMElement.prototype.addText = function (text) {
-            return this.addElem(document.createTextNode(text));
-        }
-
-        /**
-         * Returns a member element
-         * @param {int} nr
-         * @returns {*} element
-         */
-        DOMElement.prototype.getElem = function (nr) {
-            return this.myComponents[nr];
-        };
-
-        /**
-         * Adds a css class
-         * @param {string} iclss
-         * @returns {DOMElement} self
-         */
-        DOMElement.prototype.addCssClass = function (iclss) {
-            this.myClasses.push(iclss);
-            return this;
-        };
+            return handlers;
+        },
+        {});
 
-        function isDOMNode (el) {
-            return el && el.nodeName && el.nodeType
+        if (containsKey(properties, "id")) {
+            this.setID(properties.id);
         }
 
-        this._el = null;
-
-        Object.defineProperties(DOMElement.prototype, {
-            styles: {
-                get: function getStyles() {
-                    return containsKey(this.properties, "style")
-                        ? this.properties.style
-                        : {};
-                }
-            },
-            el$: {
-                get: function getElement() {
-                    if (!this._el) {
-                        // If no tagName is defined, create a fragment instead
-                        this._el = this.myType
-                            ? document.createElement(this.myType)
-                            : document.createDocumentFragment();
-                    }
-
-                    return this._el;
+        this.myClasses = properties.className ? [].concat(properties.className) : [];
+
+        // args[2] {Array} children
+        this.myComponents = args.shift(1) || [];
+    };
+
+    /**
+     * Sets the html id
+     * @param {string} iID
+     */
+    DOMElement.prototype.setID = function(iID) {
+        this.myID = iID;
+        this.addAttribute("id", iID);
+    };
+
+    /**
+     * Returns the html id
+     * @returns {string}
+     */
+    DOMElement.prototype.getID = function() {
+        return this.myID;
+    };
+
+    /**
+     * Adds a html attribute
+     * @param {string} id attribute id
+     * @param {string} content attribute content
+     * @returns {DOMElement} self
+     */
+    DOMElement.prototype.addAttribute = function(id, content) {
+        this.properties[id] = `${content}`;
```

Comment:
Just a question: is this the canonical modern way to stringify a variable, or is there another reason you used this way of working?

Reply:
There's more than one way to skin a cat. This is probably not the bestest way, I think I should have stuck with [this instead](https://exploringjs.com/impatient-js/ch_strings.html#converting-to-string) since that's what's probably happening in the background anyway.